app/controllers/change_requests_controller.rb

```diff
@@ -256,14 +255,19 @@ def persist_change(change_request)
     elsif change_request.is_a? AddressChangeRequest
       puts "AddressChangeRequest"
       address = Address.find(change_request.object_id)
-      
-      a = geocode_address address
-      unless a.nil?
-        field_change_hash["latitude"] = a.latitude
+      if address && field_change_hash["action"] == "remove"
```

Comment:
The `action` is actually a [property of the ChangeRequest object itself](https://github.com/ShelterTechSF/askdarcel-api/blob/e8fa0c49a3f224f88668313a03200a48e0c896f9/app/models/change_request.rb#L5), not the actual database object it is meant to modify. I think we should grab the `action` from the ChangeRequest object, not the field changes hash, since the latter is meant to model the fields on the actual object being modified. Since `action` is a [Rails enum field](https://api.rubyonrails.org/v5.2.5/classes/ActiveRecord/Enum.html), you should be able to write code that looks like `change_request.remove?` to check the value of that field, although this is predicated on us actually properly populating that field on the ChangeRequest object in the first place.

Also, it looks like the in the frontend code, I am sending the stringified numeric values associated with each action, not the string name nor a normal number. [There is a comment that explicitly documents this](https://github.com/ShelterTechSF/askdarcel-web/blob/c5bf7ac26ec184e7ba0f7ac701103b1f4805986f/app/pages/OrganizationEditPage.jsx#L21-L25), although I am not certain whether those come from an earlier implementation of the multiple addresses feature that never got merged. It's strange because the Postman tests always send strings names, but the frontend code always sends stringified numbers, which suggests that this field was never really being checked properly before?

Reply:
I was able to check `change_request.remove?` however I am having difficulty implementing the action specification as anything other than a field change request. Am I missing something there?

Reply:
Oh weird, I took a look at the code a little bit more closely, and I saw these lines:

https://github.com/ShelterTechSF/askdarcel-api/blob/e8fa0c49a3f224f88668313a03200a48e0c896f9/app/controllers/change_requests_controller.rb#L309-L313

It looks like we have some logic to "fall back" to just `params[:change_request]` if `params[:change_request][:field_changes]` doesn't exist, which IMO is kind of confusing. But it's not worth changing this right now, so you can probably disregard parts of my suggestion here.

I think the main drawback to this "fallback" logic is that you can't actually read from the `action` when there are `field_changes` present, but in the event where we want to `remove` something, we won't have any `field_changes` anyway, so it won't matter. This might explain, though, why there's a discrepancy between the backend expecting strings and the frontend sending numerals for the action. Knowing this now, I'm inclined to change the frontend code to send strings for the action, since it just looks like we were never reading from the `action` in the first place before.